### ISSUE TITLE
okta-react logout fails

### ISSUE THREAD

**Issue by jkupcho** [okta/okta-oidc-js#200]
Following the tutorial for react, seems like logout is failing.  I can see the local storage being cleared, but then receive a 404 on:

`DELETE https://{okta-preview}}.com/api/v1/sessions/me`

The session is actually active when I browse to the URL, and only goes inactive when I issue a delete manually through a CURL.

**Comment by jkupcho** [comment]
Also, the OPTIONS preflight call succeeds with a 200 OK, followed by the 404.

**Comment by jkupcho** [comment]
Seems related to [1961](https://github.com/okta/okta.github.io/issues/1961)

**Comment by robertjd** [comment]
Hi @jkupcho , can you link me to the tutorial you're following?  That issue says that the quickstart is out of date, it looks like the other guide is actually the one that's out of date.

Also, let's compare to the sample to find the problem: https://github.com/okta/samples-js-react/tree/master/okta-hosted-login

**Comment by jkupcho** [comment]
Hey @robertjd, thanks for replying!  So, this was actually a bit of a PEBCAK, but one that was hard to nail down.

I have the Okta extension installed in chrome for Work related stuff, and I believe the it is intercepting the dev environment requests, therefore not invalidating the correct session.

I verified my code was working on a personal laptop, then the same code wasn't working on my work.  I then opened the sample within Firefox which doesn't have the Chrome extension (obviously), and the code worked fine.

Is there a way to have the chrome extension ignore the requests?

Thanks!

**Comment by cmolodo** [comment]
I wonder if your issue in Chrome is due to 3rd-party cookies being blocked?  I had the same problem with the Angular okta-hosted-login sample app, which initially worked in Firefox but not Chrome.  (https://github.com/okta/samples-js-angular/tree/master/okta-hosted-login)

I noticed that Chrome had the little "blocked-cookies" icon and realized I had different settings for 3rd-party cookies in the two browsers.  When I started blocking 3rd party cookies in Firefox, I got the same error, and when I unblocked them in Chrome, it started working.

However, logout doesn't work in IE 11, even with 3rd-party cookies accepted.  I get the same OPTIONS request success followed by DELETE 404 error as described above.  I do notice that IE sends **only** a JSESSIONID cookie in its request where both Firefox and Chrome send back additional cookies (DT, sid, etc), not sure if that's relevant.

**Comment by amputek** [comment]
+1. This is still an issue..
`this.$auth.logout() `fails:

```
DELETE https://xxxxxxx.oktapreview.com/api/v1/sessions/me 404

errorCauses: []
errorCode: "E0000007"
errorId: "oae1WrE7cYRSk67HDn6dGBuoQ"
errorLink: "E0000007"
errorSummary: "Not found: Resource not found: me (Session)"
message: "Not found: Resource not found: me (Session)"
name: "AuthApiError"
```

I'm not using the chrome okta extension.

**Comment by jptacek** [comment]
Not sure if this helps, but I think it could be related to 3rd party cookies being blocked in the cookies. More here https://github.com/okta/samples-js-react/issues/25#issuecomment-477319277

**Comment by swiftone** [comment]
Confirmed that the current flow requires 3rd party cookies - investigating options for when they are not permitted.

**Comment by swiftone** [comment]
Two sources of this error:

1) If you don't allow 3rd party cookies, the DELETE call will fail.  (Okta is a 3rd party, being neither the user nor the app).   The error reporting is being improved per #418 , but currently 3rd party cookies are required.  

2) If you DO allow 3rd party cookies, but your Okta session expires/ends before the local application does, you will encounter the same error.  The error handling from #418 will allow a renew/re-authentication condition to be triggered.

Closing this issue with further updates appearing on the other issue, but if you have issues with permitting 3rd party cookies please open a separate issue and describe your situation.

**Comment by arminm** [comment]
@swiftone any updates on this? I'm experiencing the same error.
